services/core-web/src/components/mine/Incidents/MineIncident.js

```diff
@@ -0,0 +1,319 @@
+  fetchMineIncident,
+  updateMineIncident,
+  removeDocumentFromMineIncident,
+} from "@common/actionCreators/incidentActionCreator";
+import { clearMineIncident } from "@common/actions/incidentActions";
+import AuthorizationGuard from "@/HOC/AuthorizationGuard";
+import * as FORM from "@/constants/forms";
+import * as Permission from "@/constants/permissions";
+import Loading from "@/components/common/Loading";
+import customPropTypes from "@/customPropTypes";
+import IncidentForm from "@/components/Forms/incidents/IncidentForm";
+import ScrollSideMenu from "@/components/common/ScrollSideMenu";
+import * as routes from "@/constants/routes";
+
+const propTypes = {
+  // eslint-disable-next-line react/no-unused-prop-types
+  incident: customPropTypes.incident.isRequired,
+  createMineIncident: PropTypes.func.isRequired,
+  fetchMineIncident: PropTypes.func.isRequired,
+  updateMineIncident: PropTypes.func.isRequired,
+  clearMineIncident: PropTypes.func.isRequired,
+  removeDocumentFromMineIncident: PropTypes.func.isRequired,
+  match: PropTypes.shape({
+    params: PropTypes.shape({
+      mineGuid: PropTypes.string,
+      mineIncidentGuid: PropTypes.string,
+    }),
+  }).isRequired,
+  location: PropTypes.shape({
+    state: PropTypes.shape({
+      isEditMode: PropTypes.bool,
+      mineName: customPropTypes.mine,
+    }),
+  }).isRequired,
+  history: PropTypes.shape({ push: PropTypes.func, replace: PropTypes.func }).isRequired,
+  reset: PropTypes.func.isRequired,
+  // eslint-disable-next-line react/no-unused-prop-types
+  formValues: PropTypes.objectOf(PropTypes.any).isRequired,
+  // eslint-disable-next-line react/no-unused-prop-types
+  formIsDirty: PropTypes.bool.isRequired,
+  formErrors: PropTypes.objectOf(PropTypes.string).isRequired,
+};
+
+export class MineIncident extends Component {
+  state = {
+    isEditMode: false,
+    isNewIncident: true,
+    isLoaded: false,
+    fixedTop: false,
+  };
+
+  componentDidMount() {
+    this.handleFetchData().then(() => {
+      this.setState({ isLoaded: true, isEditMode: this.props.location.state?.isEditMode });
+    });
+    window.addEventListener("scroll", this.handleScroll);
+    this.handleScroll();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("scroll", this.handleScroll);
+    this.props.clearMineIncident();
+  }
+
+  handleScroll = () => {
+    if (window.pageYOffset > 170 && !this.state.fixedTop) {
+      this.setState({ fixedTop: true });
+    } else if (window.pageYOffset <= 170 && this.state.fixedTop) {
+      this.setState({ fixedTop: false });
+    }
+  };
+
+  handleFetchData = () => {
+    const { mineGuid, mineIncidentGuid } = this.props.match.params;
+    if (mineGuid && mineIncidentGuid) {
+      this.setState({ isNewIncident: false });
+      return this.props.fetchMineIncident(mineGuid, mineIncidentGuid);
+    }
+    return Promise.resolve();
+  };
+
+  handleCreateMineIncident = (formattedValues) => {
+    this.setState({ isLoaded: false });
```

Comment:
should this be isLoaded? in the past we have split these up between loading and submitting.

Reply:
From what I can see in the project we use submitting to disable buttons/elements during form submission and we manually set isLoaded when loading initially, creating, updating, etc.